lib/FreeRTOS-Plus-TCP/source/FreeRTOS_TCP_IP.c

```diff
@@ -1158,161 +1172,179 @@ UBaseType_t uxNewMSS;
 
 	/* The comparison with pucLast is only necessary in case the option data are
 	corrupted, we don't like to run into invalid memory and crash. */
-	while( pucPtr < pucLast )
+	xLoopAction = LOOP_CONTINUE;
+	while( ( pucPtr < pucLast ) && ( xLoopAction == LOOP_CONTINUE ) )
 	{
-		UBaseType_t xRemainingOptionsBytes = pucLast - pucPtr;
+		xLoopAction = prvCheckOptions_outerLoop( &pucPtr, &pucLast, &pxSocket, &pxTCPWindow );
+	}
+}
 
-		if( pucPtr[ 0 ] == TCP_OPT_END )
-		{
-			/* End of options. */
-			break;
-		}
-		if( pucPtr[ 0 ] == TCP_OPT_NOOP)
+/*-----------------------------------------------------------*/
+
+static FORCE_INLINE LoopAction_t prvCheckOptions_outerLoop( const unsigned char ** const pucPtr, const unsigned char ** const pucLast, FreeRTOS_Socket_t ** const pxSocket, TCPWindow_t ** const pxTCPWindow)
```

Comment:
It would be helpful to leave a comment about what the outer and inner loops are doing from a protocol perspective. Further, changing these respective function names to reflect that would be more self-documenting. 

I think that outer loop is walking the TCP header options field (i.e. which can consist of zero or more options of fixed and variable length). And I think that the inner loop is specific to parsing SACK.

Reply:
Done, please make sure I'm not saying anything daft in the comments or function names